File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
# 프로그래머스 86491 최소직사각형 LV.1
def solution(sizes):
    rows = []
    cols = []
    for size in sizes:
        if size[0] < size[1]:
            size[0], size[1] = size[1], size[0]
        rows.append(size[0])
        cols.append(size[1])
    return max(rows) * max(cols)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0, w = -1, h = -1;
        for (int i = 0; i < sizes.length; i++) {
            int max = -1, min = 1001;
            for (int j = 0; j < 2; j++) {
                max = Math.max(max, sizes[i][j]);
                min = Math.min(min, sizes[i][j]);
            }
            
            w = Math.max(w, max);
            h = Math.max(h, min);
        }
        
        return w * h;
    }
}